File: hackmate-backend/team-service/schemas/team.py
from pydantic import BaseModel
from typing import Optional, List
from datetime import datetime


class ThemeResponse(BaseModel):
    id: int
    name: str
    description: Optional[str] = None
    color_code: str
    is_active: int

    class Config:
        from_attributes = True


class TeamCreate(BaseModel):
    name: str
    idea: Optional[str] = None
    problem_statement: Optional[str] = None
    tech_skills: Optional[str] = None
    theme_id: Optional[int] = None


class TeamUpdate(BaseModel):
    name: Optional[str] = None
    idea: Optional[str] = None
    problem_statement: Optional[str] = None
    tech_skills: Optional[str] = None
    theme_id: Optional[int] = None


class TeamStatusUpdate(BaseModel):
    status: Optional[str] = None  # approved | rejected — optional for dedicated approve/reject endpoints
    floor_id: Optional[int] = None
    room_id: Optional[int] = None


class MemberResponse(BaseModel):
    id: int
    user_id: int
    team_id: int
    status: str
    joined_at: Optional[datetime] = None

    class Config:
        from_attributes = True


class TeamResponse(BaseModel):
    id: int
    name: str
    idea: Optional[str] = None
    problem_statement: Optional[str] = None
    tech_skills: Optional[str] = None
    theme_id: Optional[int] = None
    leader_id: Optional[int] = None
    leader_name: Optional[str] = None
    floor_id: Optional[int] = None
    room_id: Optional[int] = None
    floor_number: Optional[str] = None
    room_number: Optional[str] = None
    status: str
    created_at: Optional[datetime] = None
    member_count: Optional[int] = None

    class Config:
        from_attributes = True


class TeamListResponse(BaseModel):
    total: int
    teams: List[TeamResponse]


class InvitationCreate(BaseModel):
    to_user_id: int
    message: Optional[str] = None


class InvitationResponse(BaseModel):
    id: int
    team_id: int
    from_user_id: int
    to_user_id: int
    to_user_name: Optional[str] = None
    status: str
    message: Optional[str] = None
    created_at: Optional[datetime] = None

    class Config:
        from_attributes = True

File: hackmate-frontend/src/types/team.ts
export interface Theme {
  id: number
  name: string
  description?: string
  color_code: string
  is_active: number
}

export interface Team {
  id: number
  name: string
  idea?: string
  problem_statement?: string
  tech_skills?: string
  theme_id?: number
  leader_id?: number
  floor_id?: number
  room_id?: number
  status: 'pending' | 'approved' | 'rejected'
  created_at?: string
  member_count?: number
  // enriched fields (joined from other services)
  leader_name?: string
  theme_name?: string
  theme_color?: string
  floor_number?: number
  room_number?: string
}

export interface TeamMember {
  id: number
  user_id: number
  team_id: number
  status: string
  joined_at?: string
  // enriched
  name?: string
  email?: string
  tech_stack?: string
}

export interface JoinRequest {
  id: number
  user_id: number
  team_id: number
  status: 'pending' | 'approved' | 'rejected' | 'expired'
  message?: string
  created_at?: string
  responded_at?: string
  // enriched
  team_name?: string
  leader_name?: string
  name?: string
  email?: string
}

export interface Invitation {
  id: number
  team_id: number
  from_user_id: number
  to_user_id: number
  status: 'pending' | 'accepted' | 'rejected'
  message?: string
  created_at?: string
  responded_at?: string
  // enriched
  team_name?: string
  from_user_name?: string
  from_user_email?: string
  team_idea?: string
  team_problem_statement?: string
  current_members?: number
  team_members?: string
}

export interface TeamListResponse {
  total: number
  teams: Team[]
}
